### ISSUE TITLE
Positional args are handled incorrectly when -- is present

### ISSUE THREAD

**Issue by Osse** [jarro2783/cxxopts#36]
Test program:

    #include "cxxopts.hpp"
    #include <iostream>
    
    using namespace std;
    
    int main(int argc, char *argv[])
    {
        for (int i = 0; i < argc; ++i)
            cout << i << " " << argv[i] << "\n";
    
        cxxopts::Options options("posargmaster", "shows incorrect handling");
        options.add_options()
            ("dummy", "oh no", cxxopts::value<string>())
            ;
    
        options.parse(argc, argv);
    
        for (int i = 0; i < argc; ++i)
            cout << i << " " << argv[i] << "\n";
    
        return 0;
    }

Example invocations and output:

    $ ./a.out a b
    0 ./a.out
    1 a
    2 b
    0 ./a.out
    1 a
    2 b

    $ ./a.out -- a b
    0 ./a.out
    1 --
    2 a
    3 b
    0 ./a.out

It could be i am missing something, but this seems strange to me. Is this the intended behavior? If so, how should I access the positional arguments when -- is used?

**Comment by Osse** [comment]
The following works, but I assume this is not it's meant to be used:

    options.parse_positional({"", "", "positional"});
    options.parse(argc, argv);

    for(const auto& s : options["positional"].as<vector<string>>() )
        cout << s << "\n";

The two empty string can also any string not also declared as an option.

**Comment by jarro2783** [comment]
To capture the positional arguments in options you need to call `parse_positional`. Your second example is the correct way to do it, although you probably shouldn't use empty strings as option names.

I never thought of the case where you might not declare a positional option, and yet still use `--`. That part of it is probably a bug. I should probably make it leave the arguments alone after the `--` if there are no positional options declared.

**Comment by Osse** [comment]
> I should probably make it leave the arguments alone after the -- if there are no positional options declared.

This makes sense to me. The reason I didn't originally use `parse_positional()` was because the readme says "Positional arguments can be **optionally** parsed into one or more options" (my emphasis). This made me expect that the rest were left alone, and that I could do e.g. `int n = options.parse(argc, argv)` and then `argv[n]` would be the first unparsed (IOW. the first positional) argument (or maybe `int n = options.getFirstPositional()` or something along those lines; you get the idea).

Anyway, in my specific case I desired all positional arguments in one container, and using `parse_positional()` I got what I wanted :-) Thank you for this nice library.
